### ISSUE TITLE
Add "language" field to the Citation Editor

### ISSUE THREAD

**Issue by diegodlh** [diegodlh/zotero-cita#99]
As suggested by @lightgivener, one should be able to specify the language of a cited item, in the Citation Editor.

This could then be used by the Wikidata export translator to indicate the language of the title (P1476) of a Wikidata item to be created.

Related to #57

**Comment by lightgivener** [comment]
I did try and add the language in Zotero as suggested, but it won't appear in the quick statement script. I do have the same problem for place of publication and publisher at least. So a big chunk of information already included in the Zotero item info just doesn't get picked up when creating a new quick statement.

(i'll try with autocreate again later, my Wiki log in is being difficult atm.)

**Comment by diegodlh** [comment]
> I did try and add the language in Zotero as suggested, but it won't appear in the quick statement script.

I guess you mean for the citing item as suggested [here](https://github.com/diegodlh/zotero-cita/discussions/106), as doing this for the cited item is not yet supported (as per my comment above).
I've just checked Zotkat's Wikidata export translator (which Cita uses to generate QuickStatements commands) and it says that you should use "the language codes in the form as they are also used in Wikidata for monolingual text". That means that you should enter "fr" instead of "French" or "en" instead of "English" in the Zotero's item Language field. Could you try that?

> I do have the same problem for place of publication and publisher at least.

Publisher (and place of publication) are not supported because they would need to be disambiguated to their corresponding QID, and this is not supported by Zotkat's export translator yet. You can check this issue #55

**Comment by lightgivener** [comment]
Yes that worked. So use language codes is the only option for now. Thanks!

**Comment by HughP** [comment]
I see a potential problem here. The semantics of the language field in Zotero is not for the language in which the object is written in, rather it specifies the localization to be used in the citation style language export. These are distinct uses of language related information. Zotero does not have a language field which supports the language in which an object is written.

**Comment by diegodlh** [comment]
Hi, @HughP! Thanks for your comment.

> The semantics of the language field in Zotero is not for the language in which the object is written in, rather it specifies the localization to be used in the citation style language export. These are distinct uses of language related information. Zotero does not have a language field which supports the language in which an object is written.

The Zotero [documentation](https://www.zotero.org/support/kb/item_types_and_fields#general_fields) says that this field is "The language of publication for an item". However, below it says that it is used to describe what language the item's metadata is in (regardless of what language the item's content is in; to cover translated titles). This would be why this field is used by some CSL styles to decide whether [English title casing rules](https://www.zotero.org/support/kb/preventing_title_casing_for_non-english_titles) should be applied or not. As far as I know, this field is not used to specify the localization to be used by the CSL export.

> For those who use Zotero for typesetting purposes, requiring the use of the language field in this way would break their existing workflow and data set up. An approach which would not break the semantics of the existing Zotero field would be to add a Key-Value pair in Zotero's Extra field. language_of_authorship:en 

The creation of new Wikidata items in Cita uses Zotero's Wikidata QuickStatements export translator behind scenes. It is this translator that uses Zotero's language field to determine the language of the [monolingual text](https://www.wikidata.org/wiki/Special:MyLanguage/Help:Data_type#monolingualtext) value assigned to the "title" property ([P1476 ](https://www.wikidata.org/wiki/Property:P1476)). Of course, we could question (and override) this decision, but I tend to think it's correct.

As mentioned in this [thread](https://www.wikidata.org/wiki/Property_talk:P1476#Original_title_(originally_asked_in_Wikidata:Project_chat)), Wikidata's "title" property should be used to indicate official title(s) of an item. It makes sense to me that the language of an item's official title should match the item's "language of publication" specified in Zotero's language field. There seems to be a possible source of conflict regarding title translations:

On the one hand, in Wikidata, according to the thread mentioned above, for example, "each [book] translation should have it's own item with it's own title (P1476) statement". Unofficial translations should be included as labels.

In Zotero, on the other hand, the documentation mentioned above seems contradictory to me. First, it says that the language field should be used for the language of publication of an item (compatible with Wikidata's definition of language of the item's title). But then it seems to suggest that one may have a translated (unofficial?) title for an item published in another language, and that one should indicate the language of the translation in the language field. I'm not sure whether this would be the best suggestion. Where would the original (official) language be stored in this case? Shouldn't that unofficial title translation be included in the extra field instead?

> HOWEVER, I recommend looking at wikidata language related documentation for codes again. Determine if they are using ISO 639-3 (three letter codes, which covers many languages) ISO639-1 (two letter codes which only covers two dozen codes or so) OR if they are using BCP47 which is an algorithm for determining which codes to use from ISO639-1 and ISO639-3.

Apparently they are using BCP47 (aka IETF language tag): https://www.wikidata.org/wiki/Help:Monolingual_text_languages#Requirements_for_a_new_language_code

**Comment by HughP** [comment]
> The Zotero documentation says that this field is "The language of publication for an item". 
Yes, that is what the documentation says, but unfortunately the documentation does not lend itself to a plain reading (and should or could be improved for this field). From a plain reading one could write a language name, but really the only values which are of use are the ISO 639-1 codes which are supported by CSL (a sub-set of those available vis ISO 639-1). (Also note that the [Language field of software gets a different treatment as it is not the language of the User Interface, but the coding language and is matched to CSL's genre value](https://forums.zotero.org/discussion/84506/software-programing-language-and-natural-language#latest). ) One must look at what Zotero does with the data in the field. See [discussion here](https://forums.zotero.org/discussion/88201/vi-vs-vi-vn-local-codes#latest). The field specifically acts on the localization of the reference context, i.e., should ` " and  " ` be used or should  ` « and »` be used and applied by CSL style sheet. See the [CSL Locales repo](https://github.com/citation-style-language/locales), Zotero also then applies its own Capitalization rules based on the field's value — this is localization/transformation of the data, whereas CSL only localizes the punctuation and formulaic features such as `and` vs `&` vs `y` <-- in Spanish.

Language and localization is an ongoing discussion in CSL ([for example](https://discourse.citationstyles.org/t/proposal-test-condition-for-language/923)) as CSL was first designed with limited localization in mind. However JurisM, the Zotero fork maintained by Frank Bennett (who is active on the Zotero dev list and in CSL development discussions), does support multi-lingual features, via an extension to CSL. The CSL community is generally slow on the development speed. But there is a genuine interest in solving the Language related issues in CSL. My understanding is that the Zotero team is watching the CSL community to not jump the gun and make decisions first.

with regard to:
>The creation of new Wikidata items in Cita uses Zotero's Wikidata QuickStatements export translator behind scenes. It is this translator that uses Zotero's language field to determine the language of the monolingual text value assigned to the "title" property (P1476 ). Of course, we could question (and override) this decision, but I tend to think it's correct.
>
>As mentioned in this thread, Wikidata's "title" property should be used to indicate official title(s) of an item. It makes sense to me that the language of an item's official title should match the item's "language of publication" specified in Zotero's language field. There seems to be a possible source of conflict regarding title translations:

There is an issue in Zotero because users are not guided in what to do with works which are not monolingual in their content — especially their bibliographic details such as titles. This has received some attention in [CSL](https://github.com/citation-style-language/schema/issues/327).

There are many works like this. They are often seen as readers in language teaching courses, but also Bibles are often printed in two languages at the same time with verso metadata appearing in both languages. My favorite example is an academic work featuring both English and Russian. Note this is not a translation. Both languages appear in the title of this work.

> Dyakov, N. N., and A. S. Matveev, eds. 2017. Азия И Африка: Наследие И Современность — Материалы конгресса  XXIX Международный конгресс по источниковедению и историографии стран Азии и Африки (Asia and Africa: Their Heritage and Modernity — Proceedings of XXIX International Congress on Historiography and Source Studies of Asia and Africa). Vol. 2. 2 vols. Санкт-Петербург (St. Petersburg): Санкт-петербургский Государственный Университет Восточный Факультет (Saint Petersburg State University Faculty of Asian and African Studies). http://www.orienthist.spbu.ru/proceedings-of-xxix-international-congress-asia-and-africa-their-heritage-and-modernity/?lang=en.

Other cases that I have in Zotero include cases where the work I want to reference is written in Russian, where the metadata appears in Russian, but I need to use an English reference style. So, Russian cannot appear in the Zotero's language field.

Other cases include cases where the title must be transliterated to follow reference style guides such as (APA) or where a translation is also provided.

In the following case the article is written in Flemish, as commentary on an article written in English. The translation is provided by me in square brackets within Zotero's title field. Again an English reference style is used.

> Vandenhoute, Pieter Jan L. 1952. “Poro en masker: Enkele beschouwingen over ‘Masks as agents of social control in Northeast Liberia’ door Dr. G. W. Harley [Poro and mask: a few comments on masks as agents of social control in Northeast Liberia, by Dr. G.W. Harley].” Kongo-Overzee 18 (2–3): 153–98.

The following case is an book printed in Eastern Dan and French.

> Baba, Tiémoko Sébastien. 2019. Wɔ́n dhɤ: Kwɛ̏zlȁan sʌ́ʌdhɛ̋bȅ (Il y a des choses. Livre de contes). Man, Côte d’Ivoire: Pȁbhɛ̄nbhȁbhɛ̏n — EDILIS.

And this one in Japanese and French, English provided by me.
> Majima, Ichiro. 1991. “秘 密 結 社 の 語 り に み る空 間 イ マ ー ジ ュ -象 牙海岸,ダ ン族の場 合- (Les images spatiales dans les narrations de la société secrète) [Spacial images in the narratives of secret societies].” ア フ リカ研究 [Journal of African Studies] 1991 (38): 55–73. https://doi.org/10.11619/africa1964.1991.55.

My experience suggests that some works may be monolingual and have verso page metadata in the language of publication, but this is not always the case. Verso page metadata upon which reference data is usually created may not be the same language as the publication or may contain more than one language! Zotero's inability to handle this requires users to add material in languages which are not in the original work into the Zotero fields so that they appear in references correctly.
